### PR TITLE
Tweak: Improve Button widget accessibility [ED-10602]

### PR DESCRIPTION
**Pull request by rami-elementor**
If the button has a links, it should not have `role` attribute.

If the button doesn’t have a link, it can have `role` attribute.

ref: https://github.com/elementor/elementor/issues/14908

### CODE REVIEW

**Comment by rami-elementor** [comment]
The commit above fixies the issue.

The next commits are not related to a11y, they match the JS and PHP render attributes.

**Review by arielk — DISMISSED** [review]

**Comment by rami-elementor** [comment]
This PR fixes the button widget, the failed screen-shotter test is in the alert widget. Not related to this PR.

File: tests/playwright/selectors/editor-selectors.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const EditorSelectors = {
 	soundCloudIframe: 'iframe[src*="https://w.soundcloud.com/"]',
 	soundWaveForm: 'div.waveform.loaded',
 	button: {
-		getByName: ( name ) => `a[role="button"]:has-text("${ name }")`,
+		getByName: ( name ) => `a[class="elementor-button"]:has-text("${ name }")`,
```
Comment (TzviRabinovitch) [inline tests/playwright/selectors/editor-selectors.js:22]:
```suggestion
		getByName: ( name ) => `.elementor-button:has-text("Click here"),
```

**Review by TzviRabinovitch — CHANGES_REQUESTED** [review]

File: tests/playwright/selectors/editor-selectors.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ const EditorSelectors = {
 	soundCloudIframe: 'iframe[src*="https://w.soundcloud.com/"]',
 	soundWaveForm: 'div.waveform.loaded',
 	button: {
-		getByName: ( name ) => `a[class="elementor-button"]:has-text("${ name }")`,
+		getByName: ( name ) => `.elementor-button:has-text("Click here"),
```
Comment (TzviRabinovitch) [inline tests/playwright/selectors/editor-selectors.js:22]:
```suggestion
		getByName: ( name ) => `.elementor-button:has-text("Click here")`,
```

**Review by TzviRabinovitch — CHANGES_REQUESTED** [review]

**Review by TzviRabinovitch — APPROVED** [review]

**Pull request merged**